src/stream-tcp-reassemble.c

```diff
@@ -934,9 +939,9 @@ static inline bool CheckGap(TcpStream *stream, Packet *p)
 static int ReassembleUpdateAppLayer (ThreadVars *tv,
         TcpReassemblyThreadCtx *ra_ctx,
         TcpSession *ssn, TcpStream *stream,
-        Packet *p, enum StreamUpdateDir dir)
+        Packet *p, enum StreamUpdateDir dir, uint32_t gap)
 {
-    const uint64_t app_progress = STREAM_APP_PROGRESS(stream);
+    const uint64_t app_progress = STREAM_APP_PROGRESS(stream) + gap;
```

Comment:
instead of adding the gap here I think it makes more sense to update app_progress right after the GAP call to the applayer